spec/mongo/bulk_write_spec.rb

```diff
@@ -425,4 +433,44 @@
       end
     end
   end
+
+  describe 'when the collection has a validator', if: find_command_enabled? do
+
+    context 'when the documents are invalid' do
+
+      before do
+        collection_with_validator.insert_many([{:a => 1}, {:a => 2}])
+      end
+
+      let(:requests) do
+        [
+            { insert_one: { :x => 1 } },
+            { update_one: { filter: { :a => 1 },
+                            update: { '$unset' => { :a => '' }} } },
+            { replace_one: { filter: { :a => 2 },
+                             replacement: { :x => 2 }}}
+        ]
+      end
+
+      let(:result) do
+        collection_with_validator.bulk_write(requests)
+      end
+
+      let(:result2) do
+        collection_with_validator.bulk_write(
+            requests, :bypass_document_validation => true)
+      end
+
+      it 'raises BulkWriteError' do
+        expect {
+          result
+        }.to raise_exception(Mongo::Error::BulkWriteError)
+      end
+
+      it 'executes successfully when bypass_document_validation is true' do
```

Comment:
I know it seems like so many levels of contexts, but it would be more consistent to have a context 'when bypass_document_validation is true' then the it 'executes successfully' test.